networkx/classes/graph.py

```diff
@@ -315,24 +315,32 @@ def __init__(self, incoming_graph_data=None, **attr):
         {'day': 'Friday'}
 
         """
+        super().__init__()
```

Comment:
What happens if there are args or kwargs that need to be passed to the `__init__` constructor of `Mixin`?

This should probably be `super().__init__(**attr)`. And also maybe would need to change the signature of `Graph.__init__` 

PS: I'm personally not a fan of doing this. Adding a `super().__init__()` to a `object` subclass doesn't sound too fun  🤷

Reply:
> What happens if there are args or kwargs that need to be passed to the __init__ constructor of Mixin?

Yes, I thought of that, but there's nothing you can do in that case.  But at least you'll get a failure rather than silently not initializing the base class, which is much worse.

> This should probably be ...

I think it's okay to support base classes that require no arguments (which are the most prevalent class of initializers).  And to fail on the rest with a clear message at initialization time.  That's what this PR would do.

> PS: I'm personally not a fan of doing this. Adding a super().__init__() to a object subclass doesn't sound too fun

All classes are "object subclasses".  And all non-final classes should theoretically call super in their `__init__` (along with `__exit__`, `__enter__`, etc.)  These methods exhibit a pattern of extending functionality.  There are a variety of linting errors that are starting to be added to catch this potential mistake.

In any case, another benefit of this PR is that it reduces the overlapping code between Graph and Digraph, which reduces the amount of code that needs to be tested, and ensures that changes to Graph's `__init__` affects subclasses.